homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -374,7 +374,9 @@ def state_received(msg: ReceiveMessage) -> None:
             if self._config[CONF_COLOR_MODE] and "color_mode" in values:
                 self._update_color(values)
 
-            if brightness_supported(self.supported_color_modes):
+            if brightness_supported(self.supported_color_modes) or color_supported(
+                self.supported_color_modes
+            ):
```

Comment:
This is not needed, `brightness_supported` returns `True` for all color modes supporting brightness.